mmseg/models/losses/cross_entropy_loss.py

```diff
@@ -85,6 +117,16 @@ def binary_cross_entropy(pred,
             'H, W], label shape [N, H, W] are supported'
         label, weight = _expand_onehot_labels(label, weight, pred.shape,
                                               ignore_index)
+    elif avg_non_ignore:
+        # should mask out the ignored elements
+        valid_mask = ((label >= 0) & (label != ignore_index)).float()
+        if weight is not None:
+            weight *= valid_mask
+        else:
+            weight = valid_mask
```

Comment:
weight returned from `_expand_onehot_labels` has been masked

Reply:
`_expand_onehot_labels` only worked when `pred.dim() != label.dim()`. But in the case when `pred.dim() == label.dim()`, it would not be called.